main.py

```diff
@@ -59,186 +60,193 @@ def main():
+                                    'price': last_price}
+                                logger.info('Sold coins:\r\n' + str(sold_coins[coin]))
+
+                                store_order('sold.json', sold_coins)
+
+            # the buy block and logic pass
+            # announcement_coin = load_order('new_listing.json')
+            if os.path.isfile('new_listing.json'):
+                announcement_coin = load_order('new_listing.json')
+            else:
+                announcement_coin = False
 
-                    # store sold trades data
-                    else:
-                        if not test_mode:
-                            sold_coins[coin] = sell
-                            sold_coins[coin] = sell.__dict__
-                            sold_coins[coin].pop("local_vars_configuration")
+            global supported_currencies
 
-                            store_order('sold.json', sold_coins)
-                            logger.info('Order saved in sold.json')
-                        else:
-                            sold_coins[coin] = {
-                                'symbol': coin,
-                                'price': last_price,
-                                'volume': volume,
-                                'time': datetime.timestamp(datetime.now()),
-                                'profit': float(last_price) - stored_price,
-                                'relative_profit_%': round((float(
-                                    last_price) - stored_price) / stored_price * 100, 3),
-                                'id': 'test-order',
-                                'text': 'test-order',
-                                'create_time': datetime.timestamp(datetime.now()),
-                                'update_time': datetime.timestamp(datetime.now()),
-                                'currency_pair': f'{symbol}_{pairing}',
-                                'status': 'closed',
-                                'type': 'limit',
-                                'account': 'spot',
-                                'side': 'sell',
-                                'iceberg': '0',
-                                'price': last_price}
-                            logger.info('Sold coins:\r\n' + str(sold_coins[coin]))
-
-                            store_order('sold.json', sold_coins)
-
-        # the buy block and logic pass
-        # announcement_coin = load_order('new_listing.json')
-        if os.path.isfile('new_listing.json'):
-            announcement_coin = load_order('new_listing.json')
-        else:
-            announcement_coin = False
-
-        global supported_currencies
-
-        if announcement_coin not in order and announcement_coin not in sold_coins and announcement_coin not in old_coins:
-            if announcement_coin:
+            if announcement_coin and announcement_coin not in order and announcement_coin not in sold_coins and announcement_coin not in old_coins:
```

Comment:
might as well just check announcement_coin here instead of in multiple places within the scope